kotlinx-coroutines-core/common/src/flow/SharedFlow.kt

```diff
@@ -170,6 +170,15 @@ public interface MutableSharedFlow<T> : SharedFlow<T>, FlowCollector<T> {
      * Emits a [value] to this shared flow, suspending on buffer overflow if the shared flow was created
      * with the default [BufferOverflow.SUSPEND] strategy.
      *
+     * Buffer is only used by shared flow to wait for slow subscribers.
+     * **When this shared flow has no subscribers `emit` call never suspends.**
+     * It just stores the most recently emitted value to its replay cache (if it was configured),
+     * dropping the older elements from the replay cache.
+     *
+     * A shared flow configured with a [BufferOverflow] strategy other than [SUSPEND][BufferOverflow.SUSPEND]
+     * (either [DROP_OLDEST][BufferOverflow.DROP_OLDEST] or [DROP_LATEST][BufferOverflow.DROP_LATEST]) never
+     * suspends on [emit], since its buffer overflow is handled without suspension.
+     *
```

Comment:
Or, unless we want to specifically emphasize the point, maybe we could for something more succinct, like
```suggestion
     * Emits a [value] to this shared flow, suspending on buffer overflow.
     *
     * The call may never suspend unless both
     * * the [BufferOverflow] strategy is [SUSPEND][BufferOverflow.SUSPEND], and
     * * there are subscribers to this shared flow.
     *
     * If there are no subscribers, the buffer will not be used.
     * Instead, the most recently emitted value will simply be stored in
     * the replay cache (if one was configured), displacing the older elements there.
     *
```

Reply:
The combination of "never suspends" and "unless" is too hard to understand. And there's a lot of confusution about that method, so the point it to be absolutely clear. But I'll see how to make it succinct yet clear at the same time.

Reply:
Please, see how it ends up.